docker-images/couchdb-lucene/docker-entrypoint.sh

```diff
@@ -15,9 +15,21 @@ if [ ! "$COUCHDB_HOST" ]; then
     exit 1
 fi
 
+COUCHDB_USER_FILE=/run/secrets/COUCHDB_USER
+if [ -f "$COUCHDB_USER_FILE" ]; then
+    COUCHDB_USER=$(cat "$COUCHDB_USER_FILE")
+fi
 COUCHDB_PASSWORD_FILE=/run/secrets/COUCHDB_PASSWORD
-if [ "$COUCHDB_USER" ] && [ -f "$COUCHDB_PASSWORD_FILE" ]; then
-    COUCHDB_HOST="${COUCHDB_USER}:$(cat "$COUCHDB_PASSWORD_FILE")@${COUCHDB_HOST}"
+if [ -f "$COUCHDB_PASSWORD_FILE" ]; then
+    COUCHDB_PASSWORD=$(cat "$COUCHDB_PASSWORD_FILE")
+fi
+
+if [ -n "$COUCHDB_USER" -a -n "$COUCHDB_PASSWORD" ]; then
+    COUCHDB_HOST="${COUCHDB_USER}:${COUCHDB_PASSWORD}@${COUCHDB_HOST}"
+elif [ -n "$COUCHDB_USER" ]; then
+    COUCHDB_HOST="${COUCHDB_USER}@${COUCHDB_HOST}"
```

Comment:
Does CouchDB allow users without a password?

Reply:
I don't know exactly, but it doesn't hurt.